diff --git a/daemon/daemon_unix.go b/daemon/daemon_unix.go
index ce59adb..e577868 100644
--- a/daemon/daemon_unix.go
+++ b/daemon/daemon_unix.go
@@ -619,9 +619,9 @@ func configureKernelSecuritySupport(config *Config, driverName string) error {
 	if config.EnableSelinuxSupport {
 		if selinuxEnabled() {
 			// As Docker on overlayFS and SELinux are incompatible at present, error on overlayfs being enabled
-			if driverName == "overlay" {
-				return fmt.Errorf("SELinux is not supported with the %s graph driver", driverName)
-			}
+			//if driverName == "overlay" {
+			//	return fmt.Errorf("SELinux is not supported with the %s graph driver", driverName)
+			//}
 			logrus.Debug("SELinux enabled successfully")
 		} else {
 			logrus.Warn("Docker could not enable SELinux on the host system")
